src/fillTank.js

```diff
@@ -14,17 +14,38 @@
  * @param {number} amount
  */
 function fillTank(customer, fuelPrice, amount = Infinity) {
-  const { vehicle } = customer;
+  const { vehicle, money } = customer;
+
+  if (typeof customer !== 'object'
+    || customer === null || Array.isArray(customer)) {
+    throw new Error('customer should be only an object');
+  }
+
+  if ((!money && money !== 0)
+    || (!vehicle.fuelRemains && vehicle.fuelRemains !== 0)
+    || (!vehicle.maxTankCapacity || vehicle.maxTankCapacity <= 0)
+    || !vehicle) {
+    throw new Error('customer should be with proper properties');
+  }
+
+  if (isNaN(fuelPrice) || typeof fuelPrice !== 'number' || !fuelPrice) {
```

Comment:
The condition `isNaN(fuelPrice) || typeof fuelPrice !== 'number' || !fuelPrice` is redundant. Checking `typeof fuelPrice !== 'number' || fuelPrice <= 0` would be sufficient since `isNaN` is unnecessary if the type check is performed first.

Reply:
isNaN is required to check for NaN